sklearn/metrics/pairwise.py

```diff
@@ -1835,15 +1839,21 @@ def _pairwise_callable(X, Y, metric, force_all_finite=True, **kwds):
         # Calculate diagonal
         # NB: nonzero diagonals are allowed for both metrics and kernels
         for i in range(X.shape[0]):
-            x = X[i]
+            # scipy not yet implemented 1D sparse slices; can be removed and use simply
+            # `arr[ind]` once implemented
```

Comment:
```suggestion
            # scipy has not yet implemented 1D sparse slices; once implemented this can
            # be removed and `arr[ind]` can be simply used.
```